Zephyr base improvement (#23054)
zephyrproject-rtos/zephyr: doc/application/index.rst

   .. code-block:: cmake

>       # Find Zephyr, this also loads Zephyrs build system.

Reviewer: ```suggestion
      # Find Zephyr. This also loads Zephyr's build system.
```

Author: fixed

      cmake_minimum_required(VERSION 3.13.1)
      find_package(Zephyr)
      project(my_zephyr_app)

      # Add your source file to the "app" target. This must come after
      # find_package(Zephyr) which defines the target.
      target_sources(app PRIVATE src/main.c)

   ``find_package(Zephyr)`` sets the minimum CMake version and pulls in the

>    Zephyr build system, which creates a CMake target named ``app``, see

Reviewer: ```suggestion
   Zephyr build system, which creates a CMake target named ``app`` (see
```

Author: fixed

>    :ref:`cmake_pkg`. Adding sources to this target is how you include them in

Reviewer: ```suggestion
   :ref:`cmake_pkg`). Adding sources to this target is how you include them in
```

Author: fixed

   the build.

>    .. note:: ``cmake_minimum_required()`` is also invoked inside Zephyr package

Reviewer: ```suggestion
   .. note:: ``cmake_minimum_required()`` is also invoked by the Zephyr package.
```

Author: fixed

>              The most recent of the two versions will be enforced by CMake.

Reviewer: ```suggestion
              The most recent of the two versions will be enforced by CMake.
```

Author: fixed

   * As :ref:`env_vars`.
   * As a ``set(<VARIABLE> <VALUE>)`` statement in your :file:`CMakeLists.txt`

* :makevar:`ZEPHYR_BASE`: Zephyr base variable used by the build system.

>    ``find_package(Zephyr)`` will automatically set this as a cached CMake

Reviewer: It looks like there is one space too many at the beginning of this line and the next two. Since .rst is sensitive to whitespace, might be a good idea to remove them if so.

Author: fixed

   variable. But ``ZEPHYR_BASE`` can also be set as an environment variable in

>    order to enforce CMake to use a specific Zephyr installation.

Reviewer: ```suggestion
   order to force CMake to use a specific Zephyr installation.
```

Author: fixed

   found elsewhere if the CMake variable :makevar:`KCONFIG_ROOT` is
   set with an absolute path.

#. Specify that the application requires Zephyr on a new line, **after any
   lines added from the steps above**:

   .. code-block:: cmake

      find_package(Zephyr)
      project(my_zephyr_app)

   .. note:: ``find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})`` can be used if
             enforcing a specific Zephyr installation by explicitly

> 	     setting the ``ZEPHYR_BASE`` environment variable should be

Reviewer: Whitespace looks off here too.

Author: fixed

   found elsewhere if the CMake variable :makevar:`KCONFIG_ROOT` is
   set with an absolute path.

#. Specify that the application requires Zephyr on a new line, **after any
   lines added from the steps above**:

   .. code-block:: cmake

>       find_package(Zephyr)

Reviewer 2: Add a small section about backwards-compatibility with the old` include(... boilerplate)` line

Author: done
```
.. note::

include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE) is
still supported for backward compatibility with older applications.
Including boilerplate.cmake directly in the sample still requires to run
source zephyr-env.sh or execute zephyr-env.cmd before building the application.
```
